atlasdb-client/src/main/java/com/palantir/atlasdb/schema/stream/StreamStoreDefinitionBuilder.java

```diff
@@ -45,7 +45,21 @@ public StreamStoreDefinitionBuilder(String shortName, String longName, ValueType
     }
 
     public StreamStoreDefinitionBuilder hashFirstRowComponent() {
-        streamTables.forEach((tableName, streamTableBuilder) -> streamTableBuilder.hashFirstRowComponent());
+        return hashFirstNRowComponents(1);
+    }
+
+    /**
+     * We recommend that this flag is set in order to prevent hotspotting.
+     */
+    public StreamStoreDefinitionBuilder hashRowComponents() {
+        return hashFirstNRowComponents(2);
+    }
+
+    private StreamStoreDefinitionBuilder hashFirstNRowComponents(int numberOfComponentsHashed) {
+        Preconditions.checkArgument(numberOfComponentsHashed <= 2,
```

Comment:
Out of curiosity, why?

Reply:
StreamStore tables have at most two row components.

Reply:
Lets add the reason to the message in the check.